presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -510,6 +513,12 @@ protected void setup(Binder binder)
         driftServerBinder(binder).bindService(ThriftTaskService.class);
         driftServerBinder(binder).bindService(ThriftServerInfoService.class);
 
+        // Async page transport
```

Comment:
Given that we don't switch on a session property to enable/disable async page transport, why do we need to even need to seed down `asyncPageTransportEnabled` to `ExchangeClient`--can't we simply use that flag here, and if it's true do the below, to the exclusion of `TaskResource`, and if it's false do the opposite?

Reply:
Oh, I see, because `AsyncPageTransportServlet` doesn't support everything.  Perhaps it's better to enable support for session property and have both concurrently.